src/SolutionAcceleration.f90

```diff
@@ -0,0 +1,395 @@
+  CLASS(VectorType),INTENT(INOUT) :: x_new
+
+  REQUIRE(x_new%n == solver%n)
+
+  !Update iteration counter
+  solver%s=solver%s+1
+
+  IF(solver%s >= solver%start) THEN
+    !Get under-relaxed solution using mixing parameter
+    CALL BLAS_scal(x_new,solver%alpha)
+    CALL BLAS_axpy(solver%x(1),x_new,1.0_SRK-solver%alpha)
+  ENDIF
+  CALL BLAS_copy(x_new,solver%x(1))
+
+ENDSUBROUTINE step_RelaxedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Reset the initial iterate for the solver
+!> @param solver The solver to act on
+!> @param x Initial iterate solve is starting from
+!>
+SUBROUTINE reset_RelaxedPicardType(solver,x)
+  CLASS(RelaxedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x
+
+  REQUIRE(x%n == solver%n)
+
+  !Reset iteration counter
+  solver%s=0_SIK
+
+  CALL solver%setInitial(x)
+
+ENDSUBROUTINE reset_RelaxedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Clears the modified picard Type
+!> @param solver The solver to act on
+!>
+SUBROUTINE clear_ModifiedPicardType(solver)
+  CLASS(ModifiedPicardType),INTENT(INOUT) :: solver
+
+  CALL solver%clear_picardbase()
+
+  IF(ALLOCATED(solver%tmpvec)) THEN
+    CALL solver%tmpvec%clear()
+    DEALLOCATE(solver%tmpvec)
+  ENDIF
+  solver%isInit=.FALSE.
+ENDSUBROUTINE clear_ModifiedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Sets the initial vector for relaxation
+!> @param solver Solver to take step with
+!> @param x_new New solution iterate and under-relaxed / accelerated return vector
+!>
+!> This sets the initial vector into internal storage.  Also initializes solver%x
+!> and tmpvec based on the type that is passed in.
+!>
+SUBROUTINE setInitial_ModifiedPicardType(solver,x)
+  CLASS(ModifiedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x
+
+  CALL solver%setInitial_picardBase(x)
+
+  IF(.NOT. ALLOCATED(solver%tmpvec)) CALL VectorResembleAlloc(solver%tmpvec,x)
+
+ENDSUBROUTINE setInitial_ModifiedPicardType
+!
+!-------------------------------------------------------------------------------
+!> @brief Performs a single step of Modified Picard acting on the input solution vector.
+!> @param solver Solver to take step with
+!> @param x_new New solution iterate and under-relaxed / accelerated return vector
+!>
+SUBROUTINE step_ModifiedPicardType(solver,x_new)
+  CLASS(ModifiedPicardType),INTENT(INOUT) :: solver
+  CLASS(VectorType),INTENT(INOUT) :: x_new
+
+  REQUIRE(x_new%n == solver%n)
+
+  !Update iteration counter
+  solver%s=solver%s+1
+
+  CALL BLAS_copy(x_new,solver%tmpvec)
+
+  IF(solver%s >= solver%start) THEN
+    !Get under-relaxed solution using mixing parameter
+    CALL BLAS_scal(x_new,solver%alpha)
+    CALL BLAS_axpy(solver%x(1),x_new,1.0_SRK-solver%alpha)
+  ENDIF
+  CALL BLAS_copy(solver%tmpvec,solver%x(1))
```

Comment:
Do we really need `%tmpvec` on the class?  Seems like it could just be a local variable for this routine

Reply:
Because it could potentially be a large memory allocation depending on vector size... and potentially a distributed vector, I thought it would be better to store it then determine the polymorphic state of a vector and allocate it and destroy it each time.  It was potentially an optimization without a real need, but it would definitely get rid of some code.

Reply:
That's fair... I think my bigger concern is just that we seem to have required `setInitial` to be called for this class and not others, and making `tmpVec` completely local would resolve that... I could be wrong on the other classes though

Reply:
yeah, they all need `x(:)` to be allocated too... which `setInitial` also handles

Reply:
we good on this one?

Reply:
DBC checks to guard against what Aaron mentioned would be nice. Just need one on `tmpVec` I think.

Reply:
added DBC checks

Reply:
Looks good to me. @aarograh are you happy with this one?